watcher/tool/watcher/rollups.go

```diff
@@ -97,3 +106,16 @@ func receiveAndManageRollups(ctx context.Context, client *influxdb.Client, ch <-
 		}
 	}
 }
+
+func retry(ctx context.Context, fn func() error) error {
+	err := fn()
+	for err != nil {
+		select {
+		case <-ctx.Done():
```

Comment:
See above - you don't need this method.
Also, it will block here until the context has been cancelled.